surface/model_openapiv3.go

```diff
@@ -439,3 +404,58 @@ func (b *OpenAPI3Builder) buildFromSchema(name string, schema *openapiv3.Schema)
 	log.Printf("Unimplemented: could not find field info for schema: %v", schema)
 	return nil
 }
+
+// buildFromOneOfAnyOfAndAllOf adds appropriate fields to the 'schemaType' given a new 'schemaOrRef'.
+func (b *OpenAPI3Builder) buildFromOneOfAnyOfAndAllOf(schemaOrRef *openapiv3.SchemaOrReference, schemaType *Type) {
+	// Related: https://github.com/googleapis/gnostic-grpc/issues/22
+	if schema := schemaOrRef.GetSchema(); schema != nil {
+		// Build a temporary type that has the required fields; add the fields to the current schema; remove the
+		// current type
+		fieldInfo := b.buildFromSchemaOrReference("ATemporaryTypeThatWillBeRemoved", schemaOrRef)
+		if t := findType(b.model.Types, "ATemporaryTypeThatWillBeRemoved"); t != nil {
+			for _, f := range t.Fields {
+				schemaType.Fields = append(schemaType.Fields, f)
+			}
```

Comment:
Can we remove the loop and just do `schemaType.Fields = append(schemaType.Fields, t.Fields...)`?

Reply:
Done